polkadb/database_test.go

```diff
@@ -0,0 +1,117 @@
+package polkadb
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func newTestBadgerDB() (*BadgerDB, func()) {
+	dir, err := ioutil.TempDir(os.TempDir(), "badger-test")
+	fmt.Println(dir)
+	if err != nil {
+		panic("failed to create test file: " + err.Error())
+	}
+	db, err := NewBadgerDB(dir)
+	if err != nil {
+		panic("failed to create test database: " + err.Error())
+	}
+	return db, func() {
+		db.Close()
+		os.RemoveAll(dir)
+	}
+}
+
+func TestBadgerDB_PutGetDel(t *testing.T) {
+	db, remove := newTestBadgerDB()
+	defer remove()
+	testPutGetter(db, t)
+	testDelGet(db, t)
+}
+
+func testPutGetter(db Database, t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		input    string
+		expected string
+	}{
+		{"camel", "camel"},
```

Comment:
Maybe it would make sense to pass this into these two sub tests instead of declaring twice?

Reply:
Nice suggestion, I removed the redundancy by putting in a t.Run test.

Reply:
That really hurts the readability of these tests IMO, I think the original version was better